src/analysis/retail/monk/mistweaver/modules/features/CooldownThroughputTracker.tsx

```diff
@@ -46,6 +71,77 @@ class CooldownThroughputTracker extends CoreCooldownThroughputTracker {
     SPELLS.REFRESHING_JADE_WIND_HEAL.id,
     TALENTS_MONK.TRANSCENDENCE_TALENT.id,
   ];
+
+  private _hasYuLonAttribution(event: HealEvent) {
+    const tracker = this.hotTracker.hots[event.targetID]?.[event.ability.guid];
+    return tracker && this.hotTracker.fromYuLon(tracker);
+  }
+
+  private _prune(cooldown: HotAttributedTrackedEvent) {
+    cooldown.events.splice(
+      cooldown.lastAttributedIndex + 1,
+      cooldown.events.length - cooldown.lastAttributedIndex,
+    );
+    cooldown.end = cooldown.events[cooldown.events.length - 1].timestamp;
+  }
+
+  startCooldown(event: CastEvent | ApplyBuffEvent | ApplyDebuffEvent, isCastCooldown?: boolean) {
+    if (event.type === EventType.ApplyBuff && event.ability.guid === YULON_SPELL_ID) {
+      const index = this.activeCooldowns.findIndex((cooldown) => cooldown.spell === YULON_SPELL_ID);
+      if (index >= 0) {
+        const cd = this.activeCooldowns.splice(index, 1)[0];
+        this._prune(cd as HotAttributedTrackedEvent);
+      }
+    }
+
+    super.startCooldown(event, isCastCooldown);
+  }
+
+  endCooldown(event: RemoveDebuffEvent | RemoveBuffEvent) {
+    const spellId = event.ability.guid;
+    const cd = this.activeCooldowns.find((cooldown) => cooldown.spell === spellId);
+
+    super.endCooldown(event);
+
+    // Instead of ending Yu'Lon when the duration expires, keep the CD tracker
+    // active until the last hot applied during Yu'Lon falls off. In practice,
+    // we'll keep it active until the next cast or until the fight ends and prune
+    // it afterwards.
+    if (!(cd && spellId === YULON_SPELL_ID)) {
+      return;
+    }
+
+    const extendedCD = cd as HotAttributedTrackedEvent;
+    extendedCD.lastAttributedIndex = extendedCD.lastAttributedIndex ?? 0;
+    extendedCD.end = null;
+
+    this.activeCooldowns.push(extendedCD);
+  }
+
+  onFightend() {
```

Comment:
```suggestion
  onFightEnd() {
```

Reply:
this is overriding https://github.com/WoWAnalyzer/WoWAnalyzer/blob/dragonflight/src/parser/shared/modules/CooldownThroughputTracker.tsx#L212 i could rename both if you want?

Reply:
nah leave it

Reply:
that bugs me, leave same for now